docs/howto/krustlet-on-wsl2.md

```diff
@@ -0,0 +1,148 @@
+
+Krustlet requires a certificate for securing communication with the Kubernetes API. Because
+Kubernetes has its own certificates, we'll need to get a signed certificate from the Kubernetes API
+that we can use. First things first, let's create a certificate signing request (CSR):
+
+```shell
+$ mkdir -p ~/.krustlet/config
+$ cd $_
+$ openssl req -new -sha256 -newkey rsa:2048 -keyout krustlet.key -out krustlet.csr -nodes -subj "/C=US/ST=./L=./O=./OU=./CN=krustlet"
+Generating a RSA private key
+.................+++++
+....................................................+++++
+writing new private key to 'krustlet.key'
+```
+
+This will create a CSR and a new key for the certificate, using `krustlet` as the hostname of the
+server.
+
+Now that it is created, we'll need to send the request to Kubernetes:
+
+```shell
+$ cat <<EOF | kubectl apply -f -
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: krustlet
+spec:
+  request: $(cat krustlet.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+EOF
+certificatesigningrequest.certificates.k8s.io/krustlet created
+```
+
+Once that runs, you will need to approve the request:
+
+```shell
+$ kubectl certificate approve krustlet
+certificatesigningrequest.certificates.k8s.io/krustlet approved
+```
+
+After approval, you can download the cert like so:
+
+```shell
+$ kubectl get csr krustlet -o jsonpath='{.status.certificate}' | base64 --decode > krustlet.crt
+```
+
+Lastly, combine the key and the cert into a PFX bundle, choosing your own password instead of
+"password":
+
+```shell
+$ openssl pkcs12 -export -out certificate.pfx -inkey krustlet.key -in krustlet.crt -password "pass:password"
+```
+
+## Step 2: Determine the default gateway
+
+The default gateway for most Docker containers is generally `172.17.0.1`.
+This IP is only reachable, by default, from the WSL2 distro.
+However, the `eth0` host network is accessible from Windows, so we can use this IP address to connect to the WSL2 distro (where Krustlet is running). 
+
+If this was changed, check `ifconfig eth0` from
+the host OS to determine the default gateway:
+
+```console
+$ ifconfig eth0
+eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
+        inet 172.26.47.208  netmask 255.255.240.0  broadcast 172.26.47.255
+        inet6 fe80::215:5dff:fe98:ce48  prefixlen 64  scopeid 0x20<link>
+        ether 00:15:5d:98:ce:48  txqueuelen 1000  (Ethernet)
+        RX packets 16818  bytes 11576089 (11.0 MiB)
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 1093  bytes 115724 (113.0 KiB)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+In this example, I should use `172.26.47.208`.
+
+The hostname being "applied" from Windows, the default hostname will not resolve to this address, therefore you need to pass the `--node-ip` and `--node-name` flag to Krustlet.
+
+### Add a route to 172.17.0.1
```

Comment:
adding a route isn't necessary

Reply:
the route is optional indeed, like I stated, it was in case we want to have a "linux feeling".
I still have to run the demo, but I guess this means you rely on Docker Desktop doing all the ports forwarding and using `localhost` only?